docs/api.rst

```diff
@@ -370,6 +370,19 @@ Channels
     :param payload: The raw event payload data.
     :type payload: :class:`RawTypingEvent`
 
+.. function:: on_voice_channel_status_update(channel, before, after)
+
+    Called whenever the status of a voice channel has changed.
+
+    .. versionadded:: 2.4
+
+    :param channel: The channel whose status has changed.
+    :type channel: :class:`VoiceChannel`
+    :param before: The status before the update.
+    :type before: Optional[:class:`str`]
+    :param after: The status after the update.
+    :type after: Optional[:class:`str`]
```

Comment:
I think instead of an event like this we should probably make it a raw event instead with the past state potentially there, maybe as `MISSING` to differentiate it from `None` if it wasn't found at all.

Reply:
So a raw model that holds e.g. `status` and `old_status` or or more like the raw thread update event with the data payload as an attribute? Why not using a regular event with before and after status + a raw event?

Reply:
This event isn't worth having multiple. It's a waste.

I suspect you can do a raw model + `cached_status` for the old one, similar to `cached_message` in the raw message update event.